lib/libzfs/libzfs_pool.c

```diff
@@ -3733,6 +3733,13 @@ zpool_vdev_attach(zpool_handle_t *zhp, const char *old_disk,
 			(void) zpool_standard_error(hdl, errno, errbuf);
 		}
 		break;
+
+	case ZFS_ERR_ASHIFT_MISMATCH:
+		zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
+		    "The new device cannot have a higher alignment requirement "
+		    "than the top-level vdev."));
+		(void) zfs_error(hdl, EZFS_BADTARGET, errbuf);
+		break;
```

Comment:
Isn't this already covered by the default `zpool_standard_error()` message for ZFS_ERR_ASHIFT_MISMATCH?

From zpool_standard_error_fmt():
```
        case ZFS_ERR_ASHIFT_MISMATCH:
                zfs_verror(hdl, EZFS_ASHIFT_MISMATCH, fmt, ap);
                break;
```

Reply:
Yeah but vdev with lower ashift is still allowed, so it could still be misleading.
```
	case EZFS_ASHIFT_MISMATCH:
		return (dgettext(TEXT_DOMAIN, "adding devices with "
		    "different physical sector sizes is not allowed"));
```

Reply:
That's reasonable.  Just making sure I wasn't overlooking something.